From b3f40d016812ecc2db478f8b9ce5a621c78961ca Mon Sep 17 00:00:00 2001
From: Michel Thebeau <michel.thebeau@windriver.com>
Date: Wed, 28 Nov 2012 10:48:59 -0500
Subject: [PATCH] staging: cdv: rename symbol name to avoid conflict with emgd

The conflict symbol name between cdv and emgd will produce build
warnings like:

WARNING: drivers/staging/cdv/cedarview_gfx: 'PVRGetBufferClassJTable'
exported twice. Previous export was in drivers/gpu/drm/emgd/emgd.ko
WARNING: drivers/staging/cdv/cedarview_gfx: 'PVRGetDisplayClassJTable'
exported twice. Previous export was in drivers/gpu/drm/emgd/emgd.ko

Then the conflict symbols will make cedarview_gfx.ko and emgd.ko
exclusive in runtime.

Rename conflicted symbols to _cdv variant to avoid the conflict. After
the fix, we can have emgd.ko and cedarview_gfx.ko both:

[
 # lsmod
 Module                  Size  Used by
 firmware_class          5894  1 cedarview_gfx
 x_tables               13707  1 ip_tables
 ip_tables               9126  0
 ttm                    55390  1 cedarview_gfx
 emgd                  567194  0
 cedarview_gfx         382344  1
]

Signed-off-by: Liang Li <liang.li@windriver.com>
---
 drivers/staging/cdv/bc_video/bufferclass_video.c   |    2 +-
 drivers/staging/cdv/bc_video/bufferclass_video.h   |    2 +-
 .../staging/cdv/bc_video/bufferclass_video_linux.c |    4 ++--
 .../linux_framebuffer_mrst/mrstlfb_displayclass.c  |    2 +-
 .../linux_framebuffer_mrst/mrstlfb_linux.c         |    4 ++--
 .../cdv/pvr/services4/include/kernelbuffer.h       |    2 +-
 .../cdv/pvr/services4/include/kerneldisplay.h      |    2 +-
 .../cdv/pvr/services4/srvkm/common/deviceclass.c   |    4 ++--
 .../cdv/pvr/services4/srvkm/env/linux/module.c     |    4 ++--
 9 files changed, 13 insertions(+), 13 deletions(-)

diff --git a/drivers/staging/cdv/bc_video/bufferclass_video.c b/drivers/staging/cdv/bc_video/bufferclass_video.c
index 3464b46..9e6ee15 100644
--- a/drivers/staging/cdv/bc_video/bufferclass_video.c
+++ b/drivers/staging/cdv/bc_video/bufferclass_video.c
@@ -199,7 +199,7 @@ BC_Video_Register (int id)
             return (BCE_ERROR_INIT_FAILURE);
         }
         if (BCGetLibFuncAddr
-                (psDevInfo->hPVRServices, "PVRGetBufferClassJTable",
+                (psDevInfo->hPVRServices, "PVRGetBufferClassJTable_cdv",
                  &pfnGetPVRJTable) != BCE_OK)
         {
             return (BCE_ERROR_INIT_FAILURE);
diff --git a/drivers/staging/cdv/bc_video/bufferclass_video.h b/drivers/staging/cdv/bc_video/bufferclass_video.h
index 07dd33a..b681bf2 100644
--- a/drivers/staging/cdv/bc_video/bufferclass_video.h
+++ b/drivers/staging/cdv/bc_video/bufferclass_video.h
@@ -60,7 +60,7 @@ extern "C"
     } bc_buf_params_t;
 
     extern IMG_IMPORT IMG_BOOL
-        PVRGetBufferClassJTable (PVRSRV_BC_BUFFER2SRV_KMJTABLE * psJTable);
+        PVRGetBufferClassJTable_cdv (PVRSRV_BC_BUFFER2SRV_KMJTABLE * psJTable);
 
 #define BC_VIDEO_DEVICE_MAX_ID          5
 #define BC_CAMERA_DEVICEID              8
diff --git a/drivers/staging/cdv/bc_video/bufferclass_video_linux.c b/drivers/staging/cdv/bc_video/bufferclass_video_linux.c
index f70418a..4c30202 100644
--- a/drivers/staging/cdv/bc_video/bufferclass_video_linux.c
+++ b/drivers/staging/cdv/bc_video/bufferclass_video_linux.c
@@ -415,12 +415,12 @@ BCE_ERROR
 BCGetLibFuncAddr (BCE_HANDLE unref__ hExtDrv, char *szFunctionName,
         PFN_BC_GET_PVRJTABLE * ppfnFuncTable)
 {
-    if (strcmp ("PVRGetBufferClassJTable", szFunctionName) != 0)
+    if (strcmp ("PVRGetBufferClassJTable_cdv", szFunctionName) != 0)
     {
         return (BCE_ERROR_INVALID_PARAMS);
     }
 
-    *ppfnFuncTable = PVRGetBufferClassJTable;
+    *ppfnFuncTable = PVRGetBufferClassJTable_cdv;
 
     return (BCE_OK);
 }
diff --git a/drivers/staging/cdv/pvr/services4/3rdparty/linux_framebuffer_mrst/mrstlfb_displayclass.c b/drivers/staging/cdv/pvr/services4/3rdparty/linux_framebuffer_mrst/mrstlfb_displayclass.c
index 9479680..7ea2cb7 100644
--- a/drivers/staging/cdv/pvr/services4/3rdparty/linux_framebuffer_mrst/mrstlfb_displayclass.c
+++ b/drivers/staging/cdv/pvr/services4/3rdparty/linux_framebuffer_mrst/mrstlfb_displayclass.c
@@ -1399,7 +1399,7 @@ MRST_ERROR MRSTLFBInit(struct drm_device * dev)
 			return (MRST_ERROR_INIT_FAILURE);
 		}
 
-		if(MRSTLFBGetLibFuncAddr ("PVRGetDisplayClassJTable", &pfnGetPVRJTable) != MRST_OK)
+		if(MRSTLFBGetLibFuncAddr ("PVRGetDisplayClassJTable_cdv", &pfnGetPVRJTable) != MRST_OK)
 		{
 			return (MRST_ERROR_INIT_FAILURE);
 		}
diff --git a/drivers/staging/cdv/pvr/services4/3rdparty/linux_framebuffer_mrst/mrstlfb_linux.c b/drivers/staging/cdv/pvr/services4/3rdparty/linux_framebuffer_mrst/mrstlfb_linux.c
index d33bfad..95c4890 100644
--- a/drivers/staging/cdv/pvr/services4/3rdparty/linux_framebuffer_mrst/mrstlfb_linux.c
+++ b/drivers/staging/cdv/pvr/services4/3rdparty/linux_framebuffer_mrst/mrstlfb_linux.c
@@ -74,13 +74,13 @@ void MRSTLFBFreeKernelMem(void *pvMem)
 
 MRST_ERROR MRSTLFBGetLibFuncAddr (char *szFunctionName, PFN_DC_GET_PVRJTABLE *ppfnFuncTable)
 {
-	if(strcmp("PVRGetDisplayClassJTable", szFunctionName) != 0)
+	if(strcmp("PVRGetDisplayClassJTable_cdv", szFunctionName) != 0)
 	{
 		return (MRST_ERROR_INVALID_PARAMS);
 	}
 
 
-	*ppfnFuncTable = PVRGetDisplayClassJTable;
+	*ppfnFuncTable = PVRGetDisplayClassJTable_cdv;
 
 	return (MRST_OK);
 }
diff --git a/drivers/staging/cdv/pvr/services4/include/kernelbuffer.h b/drivers/staging/cdv/pvr/services4/include/kernelbuffer.h
index 4cd36d2..7b332b0 100644
--- a/drivers/staging/cdv/pvr/services4/include/kernelbuffer.h
+++ b/drivers/staging/cdv/pvr/services4/include/kernelbuffer.h
@@ -63,7 +63,7 @@ typedef struct PVRSRV_BC_BUFFER2SRV_KMJTABLE_TAG
 
 typedef IMG_BOOL (*PFN_BC_GET_PVRJTABLE) (PPVRSRV_BC_BUFFER2SRV_KMJTABLE); 
 
-IMG_IMPORT IMG_BOOL PVRGetBufferClassJTable(PVRSRV_BC_BUFFER2SRV_KMJTABLE *psJTable);
+IMG_IMPORT IMG_BOOL PVRGetBufferClassJTable_cdv(PVRSRV_BC_BUFFER2SRV_KMJTABLE *psJTable);
 
 #if defined (__cplusplus)
 }
diff --git a/drivers/staging/cdv/pvr/services4/include/kerneldisplay.h b/drivers/staging/cdv/pvr/services4/include/kerneldisplay.h
index cdbef00..7498118 100644
--- a/drivers/staging/cdv/pvr/services4/include/kerneldisplay.h
+++ b/drivers/staging/cdv/pvr/services4/include/kerneldisplay.h
@@ -154,7 +154,7 @@ typedef struct DISPLAYCLASS_FLIP_COMMAND_TAG
 
 typedef IMG_BOOL (*PFN_DC_GET_PVRJTABLE)(PPVRSRV_DC_DISP2SRV_KMJTABLE);
 
-IMG_IMPORT IMG_BOOL PVRGetDisplayClassJTable(PVRSRV_DC_DISP2SRV_KMJTABLE *psJTable);
+IMG_IMPORT IMG_BOOL PVRGetDisplayClassJTable_cdv(PVRSRV_DC_DISP2SRV_KMJTABLE *psJTable);
 
 
 #if defined (__cplusplus)
diff --git a/drivers/staging/cdv/pvr/services4/srvkm/common/deviceclass.c b/drivers/staging/cdv/pvr/services4/srvkm/common/deviceclass.c
index 13bfa8d..044d3ee 100644
--- a/drivers/staging/cdv/pvr/services4/srvkm/common/deviceclass.c
+++ b/drivers/staging/cdv/pvr/services4/srvkm/common/deviceclass.c
@@ -1675,7 +1675,7 @@ IMG_VOID IMG_CALLCONV PVRSRVSetDCState(IMG_UINT32 ui32State)
 
 
 IMG_EXPORT
-IMG_BOOL PVRGetDisplayClassJTable(PVRSRV_DC_DISP2SRV_KMJTABLE *psJTable)
+IMG_BOOL PVRGetDisplayClassJTable_cdv(PVRSRV_DC_DISP2SRV_KMJTABLE *psJTable)
 {
 	psJTable->ui32TableSize = sizeof(PVRSRV_DC_DISP2SRV_KMJTABLE);
 	psJTable->pfnPVRSRVRegisterDCDevice = &PVRSRVRegisterDCDeviceKM;
@@ -2000,7 +2000,7 @@ PVRSRV_ERROR PVRSRVGetBCBufferKM (IMG_HANDLE hDeviceKM,
 
 
 IMG_EXPORT
-IMG_BOOL PVRGetBufferClassJTable(PVRSRV_BC_BUFFER2SRV_KMJTABLE *psJTable)
+IMG_BOOL PVRGetBufferClassJTable_cdv(PVRSRV_BC_BUFFER2SRV_KMJTABLE *psJTable)
 {
 	psJTable->ui32TableSize = sizeof(PVRSRV_BC_BUFFER2SRV_KMJTABLE);
 
diff --git a/drivers/staging/cdv/pvr/services4/srvkm/env/linux/module.c b/drivers/staging/cdv/pvr/services4/srvkm/env/linux/module.c
index 9899d85..cfd6b61 100644
--- a/drivers/staging/cdv/pvr/services4/srvkm/env/linux/module.c
+++ b/drivers/staging/cdv/pvr/services4/srvkm/env/linux/module.c
@@ -123,8 +123,8 @@ MODULE_PARM_DESC(gPVRDebugLevel, "Sets the level of debug output (default 0x7)")
 #endif 
 
  
-EXPORT_SYMBOL(PVRGetDisplayClassJTable);
-EXPORT_SYMBOL(PVRGetBufferClassJTable);
+EXPORT_SYMBOL(PVRGetDisplayClassJTable_cdv);
+EXPORT_SYMBOL(PVRGetBufferClassJTable_cdv);
 
 #if defined(PVR_LDM_MODULE)
 static struct class *psPvrClass;
-- 
1.7.9.7

